homeassistant/components/sensor/airvisual.py

```diff
@@ -0,0 +1,304 @@
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        yield from super().async_update()
+        aqi = self._data.pollution_info.get('aqi{0}'.format(self._locale))
+
+        try:
+            [level] = [
+                i for i in POLLUTANT_LEVEL_MAPPING
+                if i['minimum'] <= aqi <= i['maximum']
+            ]
+            self._state = level.get('label')
+        except ValueError:
+            self._state = None
+
+
+class AirQualityIndexSensor(AirVisualBaseSensor):
+    """Define a sensor to measure AQI."""
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        yield from super().async_update()
+        self._state = self._data.pollution_info.get(
+            'aqi{0}'.format(self._locale))
+
+
+class MainPollutantSensor(AirVisualBaseSensor):
+    """Define a sensor to the main pollutant of an area."""
+
+    def __init__(self, data, name, icon, locale):
+        """Initialize."""
+        super().__init__(data, name, icon, locale)
+        self._symbol = None
+        self._unit = None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._data:
+            return merge_two_dicts(super().device_state_attributes, {
+                ATTR_POLLUTANT_SYMBOL: self._symbol,
+                ATTR_POLLUTANT_UNIT: self._unit
+            })
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        yield from super().async_update()
+        symbol = self._data.pollution_info.get('main{0}'.format(self._locale))
+        pollution_info = POLLUTANT_MAPPING.get(symbol, {})
+        self._state = pollution_info.get('label')
+        self._unit = pollution_info.get('unit')
+        self._symbol = symbol
+
+
+class AirVisualData(object):
+    """Define an object to hold sensor data."""
+
+    def __init__(self, client, latitude, longitude, radius):
+        """Initialize."""
+        self._city = None
+        self._client = client
+        self._country = None
+        self._latitude = latitude
+        self._longitude = longitude
+        self._pollution_info = None
+        self._radius = radius
+        self._state = None
+
+    @property
+    def client(self):
+        """Define a property to access the pyairvisual client."""
+        return self._client
+
+    @property
+    def city(self):
+        """Define a property to access the city."""
+        return self._city
+
+    @property
+    def country(self):
+        """Define a property to access the country."""
+        return self._country
+
+    @property
+    def pollution_info(self):
+        """Define a property to access the pollution information."""
+        return self._pollution_info  # pylint: disable=syntax-error
+
+    @property
+    def state(self):
+        """Define a property to access the state."""
+        return self._state
```

Comment:
Consider making all these properties regular instance attributes instead, since you don't seem to need any more logic than returning the attribute.

Reply:
Will do!